### ISSUE TITLE
SelectableText highlight does not work correctly when canRequestFocus=false

### ISSUE THREAD

**Issue by esDotDev** [flutter/flutter#77721]
Currently by default, SelectableText seems to set `canRequestFocus=true`, this generally not expected behavior on web/desktop where you would expect to skip read-only text fields when using Tab traversal.

When trying to disable it manually, (passing our own focus node, w/ canRequestFocus: false), we get un-intended side effects, where multiple SelectableText can now be selected, and they never lose their focus:
![image](https://user-images.githubusercontent.com/736973/110510874-db85fe80-80c0-11eb-80c3-86fc3c114f4b.png)

We would like some succinct way of getting web-like behavior of selectable, but not focusable, text.

**Comment by esDotDev** [comment]
Highlight one of the selectable-texts, then highlight the other, then enter the text field:
```
return Column(children: [
  SelectableText("Hello", focusNode: FocusNode(canRequestFocus: false)),
  ExcludeFocus(child: SelectableText("World")),
  TextFormField(),
]);
```
You should now have 2 incorrectly highlighted text areas. 

```
Flutter 2.0.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision c5a4b4029c (6 days ago) • 2021-03-04 09:47:48 -0800
Engine • revision 40441def69
Tools • Dart 2.12.0
```

**Comment by esDotDev** [comment]
I mentioned in the original post " you would expect to **skip read-only text fields when using Tab traversal**."

With default behavior, if you hit "Tab" on Web or Desktop platforms, it will take 3 taps to get back to the text field, because each of the `SelectableText` will get focused.

Generally we don't want "read only" text to have this behavior. If you look at your typical web page, you can not tab between paragraphs for example.

**Comment by pedromassangocode** [comment]
> I mentioned in the original post " you would expect to **skip read-only text fields when using Tab traversal**."
> 
> With default behavior, if you hit "Tab" on Web or Desktop platforms, it will take 3 taps to get back to the text field, because each of the `SelectableText` will get focused.

I think there is a better way to get this behavior by explicitly saying to the focus traversal policy to skip those widgets. You achieve this by setting `skipTraversal` to `true` in the `focusNode`. I was able to achieve your expected behavior with the code sample bellow:

```dart
import 'package:flutter/material.dart';

void main() => runApp(MaterialApp(title: 'Flutter', home: MyHomePage(),));

class MyHomePage extends StatelessWidget {
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            //SelectableText("Hello", focusNode: FocusNode(canRequestFocus: false),),
            //ExcludeFocus(child: SelectableText("World")),

            SelectableText("Hello", focusNode: FocusNode(skipTraversal: true),),
            SelectableText("World", focusNode: FocusNode(skipTraversal: true),),
            TextFormField(),
          ],
        ),
      ),
    );
  }
}
```

### skipTraversal docs

![Screen Shot 2021-03-11 at 07 57 27](https://user-images.githubusercontent.com/67424416/110747991-6e4ca780-823f-11eb-97c7-4958c5f7174a.png)

**Comment by esDotDev** [comment]
Thanks, that does work! 

Seems like there is still a bug here, you would never expect the text to be de-selected. And it's also inconsistent with `FocusNode(skipTraversal: true)` not causing this issue, but  `FocusNode(canRequestFocus: false)` and `ExcludeFocus` both causing it. 

This unblocks us for now tho, thanks!

**Comment by pedromassangocode** [comment]
> Seems like there is still a bug here,

This seems to be a different issue, can you please file a new issue (filling the issue template) to discuss it? We don't track more than one problem in the same GitHub issue.
Thank you

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by esDotDev** [comment]
I have opened a higher level issue here, just asking for the correct default behavior: https://github.com/flutter/flutter/issues/79945
